appserver/monitoring-console/webapp/src/main/webapp/js/mc-view-components.js

```diff
@@ -221,65 +229,106 @@ MonitoringConsole.View.Components = (function() {
       return { onCreation: onCreation };
    })();
 
-   /**
-   * Component to navigate pages. More a less a dropdown.
+  /**
+   * Component drawn for each widget legend item to indicate data status.
    */
-   let Navigation = (function() {
+  let Indicator = (function() {
 
-      function onUpdate(model) {
-         let dropdown = $('<select/>');
-         dropdown.change(() => model.onChange(dropdown.val()));
-         for (let i = 0; i < model.pages.length; i++) {
-            let pageModel = model.pages[i];
-            dropdown.append($('<option/>', {value: pageModel.id, text: pageModel.label, selected: pageModel.active }));
-            if (pageModel.active) {
-               dropdown.val(pageModel.id);
-            }
-         }
-         let nav = $("#Navigation"); 
-         nav.empty();
-         nav.append(dropdown);
-         return dropdown;
-      }
-
-      return { onUpdate: onUpdate };
-   })();
+    function onCreation(model) {
+       if (!model.text) {
+          return $('<div/>', {'class': 'Indicator', style: 'display: none;'});
+       }
+       let html = model.text.replace(/\*([^*]+)\*/g, '<b>$1</b>').replace(/_([^_]+)_/g, '<i>$1</i>');
+       return $('<div/>', { 'class': 'Indicator status-' + model.status }).html(html);
+    }
 
+    return { onCreation: onCreation };
+  })();
 
-   let Indicator = (function() {
+  /**
+   * Component for any of the text+icon menus/toolbars.
+   */
+  let Menu = (function() {
 
-      function onCreation(model) {
-         if (!model.text) {
-            return $('<div/>', {'class': 'Indicator', style: 'display: none;'});
-         }
-         let html = model.text.replace(/\*([^*]+)\*/g, '<b>$1</b>').replace(/_([^_]+)_/g, '<i>$1</i>');
-         return $('<div/>', { 'class': 'Indicator status-' + model.status }).html(html);
+    function onCreation(model) {
+      let attrs = { 'class': 'Menu' };
+      if (model.id)
+        attrs.id = model.id;
```

Comment:
Some of the if statements look short enough to fit onto a single line, which I personally find more readable, just another nitpick

Reply:
I opted to not do single line ifs since I also opted to not always do brackets so I made this compromise. Overall I still in the process of finding good format rules for the JS so it is not entirely consistent either but this is good feedback I definitely will consider in the future.